request_handler_static.cpp

```diff
@@ -20,52 +21,70 @@ const std::map<std::string, std::string> MIMEmap  = {
 };
 
 
-StaticHandler::StaticHandler(std::string root_dir) : _root(root_dir){
+RequestHandler::Status StaticHandler::Init(const std::string& uri_prefix, const NginxConfig& config){
+  _uri_prefix = uri_prefix;
+  _config = config;
+  _root = "";
+
+  for (auto statement : config.statements_) {
+    if (statement->tokens_[0] == "root" && statement->tokens_.size() > 1)
+      _root = statement->tokens_[1];
+  }   
+
+  if (_root == "")
+    return RequestHandler::ERROR;
+
+  return RequestHandler::OK;
 }
 
-bool StaticHandler::handle_request(const HttpRequest& request, HttpResponse* &response){
+RequestHandler::Status StaticHandler::HandleRequest(const Request& request, Response* response){
   std::stringstream data;
+  
+  std::string uri = request.uri();
+ 
+  std::size_t slash_pos = uri.find("/", 1);
```

Comment:
Again this file_path extraction should probably be better as there are configurations that can break this (I believe).